compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/NemoOptimizer.java

```diff
@@ -46,19 +49,27 @@
 public final class NemoOptimizer implements Optimizer {
   private final String dagDirectory;
   private final Policy optimizationPolicy;
+  private final String environmentTypeStr;
+  private final ClientRPC clientRPC;
 
   private final Map<UUID, Integer> cacheIdToParallelism = new HashMap<>();
   private int irDagCount = 0;
 
 
   /**
-   * @param dagDirectory to store JSON representation of intermediate DAGs.
-   * @param policyName   the name of the optimization policy.
+   * @param dagDirectory       to store JSON representation of intermediate DAGs.
+   * @param policyName         the name of the optimization policy.
+   * @param environmentTypeStr the environment type of the workload to optimize the DAG for.
+   * @param clientRPC          the RPC channel to communicate with the client.
    */
   @Inject
   private NemoOptimizer(@Parameter(JobConf.DAGDirectory.class) final String dagDirectory,
```

Comment:
Is this compile time or runtime optimizer?

Reply:
It handles both